_posixsubprocess.c

```diff
@@ -154,6 +154,29 @@ _is_fd_in_sorted_fd_sequence(int fd, PyObject *fd_sequence)
     return 0;
 }
 
+static int
+make_inheritable(PyObject *py_fds_to_keep, int errpipe_write)
+{
+    Py_ssize_t i, len;
+
+    len = PySequence_Length(py_fds_to_keep);
+    for (i = 0; i < len; ++i) {
+        PyObject* fdobj = PySequence_Fast_GET_ITEM(py_fds_to_keep, i);
+        long fd = PyLong_AsLong(fdobj);
+        assert(!PyErr_Occurred());
+        assert(0 <= fd && fd <= INT_MAX);
+        if (fd == errpipe_write) {
+            /* errpipe_write is part of py_fds_to_keep. It must be closed at
+               exec(), but kept open in the child process until exec() is
+               called. */
+            continue;
+        }
+        if (set_inheritable((int)fd, 1, 1, NULL) < 0)
```

Comment:
Thanks for the feedback.

I see that it makes sense to remove `atomic_flag_works` and simplify the conditional logic that uses it, since the only call site passes `NULL` for it.

I don't understand why an exception cannot be raised. I'd like to understand the reason for that. Is it because `child_exec` expects `-1` to be returned for any error case, so that execution jumps to the `error` label?